src/context/MetricsContext/errorReporting.ts

```diff
@@ -1,18 +1,86 @@
-import { getAccount } from "@wagmi/core";
 import CustomErrors from "abis/CustomErrors.json";
 import { isDevelopment, isLocal } from "config/env";
 import cryptoJs from "crypto-js";
 import { extractDataFromError } from "domain/synthetics/orders/simulateExecuteOrderTxn";
 import { OracleFetcher, useOracleKeeperFetcher } from "domain/synthetics/tokens";
 import { ethers } from "ethers";
-import { useEffect } from "react";
-import { extractError } from "./contracts/transactionErrors";
-import { rainbowKitConfig } from "./wallets/rainbowKitConfig";
-import { useLocalStorageSerializeKey } from "./localStorage";
+import { useCallback, useEffect, useMemo, useRef } from "react";
+import { extractError } from "../../lib/contracts/transactionErrors";
+import { useLocalStorageSerializeKey } from "../../lib/localStorage";
 import { SHOW_DEBUG_VALUES_KEY } from "config/localStorage";
+import { useSubaccountAddress } from "context/SubaccountContext/SubaccountContext";
+import useIsMetamaskMobile from "../../lib/wallets/useIsMetamaskMobile";
+import { useChainId } from "../../lib/chains";
+import { getWalletNames } from "../../lib/wallets/getWalletNames";
+import { getAppVersion } from "../../lib/version";
 
 const IGNORE_ERROR_MESSAGES = ["user rejected action", "failed to fetch"];
 
+export function useUiMetrics() {
+  const { chainId } = useChainId();
+  const fetcher = useOracleKeeperFetcher(chainId);
+  const subaccountAddress = useSubaccountAddress();
+  const isMetamaskMobile = useIsMetamaskMobile();
+
+  const pendingEvents = useRef({});
+  const timers = useRef({});
+
+  const setPendingEvent = useCallback((key: string, eventData: any) => {
```

Comment:
I think eventData should be typed based on metricType, e.x. `MetricEventData = OrderIncreaseMetricEventData | OrderDecreaseMetricEventData | etc`